### PR TITLE
GSoC 2017 Proposal - Data Visualization For Archived Content

### PR DESCRIPTION
**Pull request by Blackglade**
@b5 Here is the proposal for my idea that we discussed on slack!

Relevant other people I should probably tag: @dcwalk @titaniumbones 

Also, incase you don't want to read this markdown file, I also have an adjacent Google doc with the write up available as well! Find it here: https://docs.google.com/document/d/11iNZW0QVtRH8JasixzObAs7pMLOuOH9V3NJJUY-WPJM/edit?usp=sharing
Let me know what you think! 

***Note: I purposely left it a bit vague mostly because I wanted some feedback before I chose a definite model/graph to work on!***

### CODE REVIEW

**Comment by dcwalk** [comment]
Thanks for sharing @Blackglade! I'll be able to take a look at this tonight.

File: gsoc/proposal-Data-Visualization-For-Archived-Content.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+**Describe how your project meets this need:**  
+This is where my project comes in. Using data visualization libraries (such as https://d3js.org/), my project’s goal will be to develop a set of models/graphs accurately displaying any “meaningful” data you have archived. I would ideally be using the DataRefuge API for this project as a means to gather data for my in-browser model, but could eventually directly integrate with your toolkit to make API calls. These visualizations could be added directly on top of your React.js app view layer (https://www.archivers.space/). Working with my assigned mentors to identify how this data is “relevant” and then constructing an appropriate model to display it would allow individuals visiting your app to gain a better understanding of the data being collected. This would be a preferential solution, as opposed to sifting through tons of lists.
```
Comment (b5) [inline gsoc/proposal-Data-Visualization-For-Archived-Content.md]: If accepted we would expect a fairly tight integration between react & d3 for performance reasons for end users. As a suggested workflow you could first build the graphs in pure d3js, and we'd then guide you through the steps for integrating with react's rendering engine, using keeping the underlying d3-core layout functions & your d3-oriented code. We would make sure you're fully supported during this process.

File: gsoc/proposal-Data-Visualization-For-Archived-Content.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+**Milestones/Timeline:**  
+This project is essentially composed of **3 major milestones**: working with my mentors to identify which data is meaningful and worth developing a model for, developing the model while simultaneously receiving feedback on improvements, and then eventually committing it to production for use. Below you will find a rough timeline:
```
Comment (b5) [inline gsoc/proposal-Data-Visualization-For-Archived-Content.md]: These milestones seem great, the only change I'd advise is we'd ask ask you to commit continuously to production, at first with a minimum viable version of an initial visualization, and then continuously improve it over time before moving on to the next. Your third milestone could instead be documenting what you learned for the greater community to read about.

File: gsoc/proposal-Data-Visualization-For-Archived-Content.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+* **Aug 8th - Aug 22nd**
+Reach out to individuals using the website as well as my mentors to see if I can identify any other changes to further optimize the experience and make it more interactive and meaningful on the end-user. As with any product being shipped, testing usage and interaction with the product is important to gauge its overall perception. 
+
+* **Aug 22nd - End Of Program**
+Final communication with mentors and hopefully continue to work and further develop/refine models in the future with EDGI.
+
+
+**Deliverables:**  
+A set of models/graphs for visualization archived data (I say a set instead of a specific number, because I am not too sure going in what data we would want to display. Realistically I would like to work on from anywhere between 3 to 5 models).
```
Comment (b5) [inline gsoc/proposal-Data-Visualization-For-Archived-Content.md]: This is great! The only thing I'd add is a final recap document on what you'd learned and how the community came to benefit from your work (as mentioned above)

File: gsoc/proposal-Data-Visualization-For-Archived-Content.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+**Teamwork:**
+In regards to teamwork experience, I was the head programmer in my high school robotics team and during competition season I worked with other programmers to properly delegate tasks and come up with a timeline for delivering code. It involved a lot of management, but also a lot of time making sure that everyone on the team agreed with the coding-style guidelines I put forward. 
+
+**Interests:**
+Specific to the field of Computer Science, while I have a lot of experience with full-stack development, I’ve recently been more interested in getting involved with low-level architecture (such as embedded systems and signal processing). Outside of CS, I love reading, playing my ukulele, chess, and of course getting a good night’s sleep!
+
+**Commitment:**
+In regards to commitment, I will try to dedicate about 30 hours of my week focused on this task. I have no other commitments for this summer so far, nor any vacation plans. I will be able to fully focus on this project.
+
+### Thank you for taking the time to reading my proposal and I hope you will consider me for GSoC 2017!
```
Comment (b5) [inline gsoc/proposal-Data-Visualization-For-Archived-Content.md:94]: Thank you for submitting!

**Review by b5 — COMMENTED** [review]
This is great Harsh! I left a few small comments that you may wish to use to alter your proposal, but the general outline of what you've proposed here looks great to me. Thanks for submitting!

File: gsoc/proposal-Data-Visualization-For-Archived-Content.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+* **Jul 11th -  Jul 25th**
+I will have mostly completed any models at this time and will submit it to my mentors for review. If any changes and/or any new models need to be made, I will work on them as well! Most of my time this period will be spent communicating with the team and listening to any feedback they have.
+
+* **Jul 25th - Aug 8th**
+Finalize any ideas I had and hopefully commit my developed models to the “live” website 
+
+* **Aug 8th - Aug 22nd**
+Reach out to individuals using the website as well as my mentors to see if I can identify any other changes to further optimize the experience and make it more interactive and meaningful on the end-user. As with any product being shipped, testing usage and interaction with the product is important to gauge its overall perception.
```
Comment (suchthis) [inline gsoc/proposal-Data-Visualization-For-Archived-Content.md]: Thanks for this thoughtful inclusion! I love that you've planned ahead for a round of post-deploy refinement based on user feedback.

Comment (mhucka) [reply gsoc/proposal-Data-Visualization-For-Archived-Content.md]: I also appreciate the explanations of what is expected during each time period. All too often, proposals have only a bullet list and leaves too much to the imagination.

**Review by mhucka — COMMENTED** [review]
Minor comments

**Comment by Blackglade** [comment]
Thank you @mhucka @suchthis for your comments 😄 !

@b5  The initial plan was to finish one model, before moving onto the next one, but I wasn't too sure how to properly write that into a timeline format split up into distinctive sections. Anyway, I've updated the proposal to make sure I specified that I will be doing exactly that. I've also changed my milestones so I'm committing continuously and have a part of the timeline dedicated to documenting what I learned. The only parts that really changed is the **Describe how your project meets this need**, **Milestones/Timeline** , and **Deliverables** section.

Thanks again for your feedback and let me know if there is anything else I could change before submitting it on the website (don't really want to wait until the last day to get this in). Thanks again for your help.

Oh and as always, if you don't want the markdown file, there is an updated version on the google doc link: https://docs.google.com/document/d/11iNZW0QVtRH8JasixzObAs7pMLOuOH9V3NJJUY-WPJM/edit

**Comment by suchthis** [comment]
Thanks for submitting an update to your proposal, @Blackglade! It's really nice to see you incorporating the feedback to include continuous integration and a documentation step. @b5 will have a better perspective on whether to further update your timeline, e.g. whether your allotment of time for information gathering vs. model building seems accurate. If you would still like to build into the proposal the idea that you'll be working on models sequentially, you could just add that you'll be defining a set of prioritized models, and then working during the model-delivery period on each model in sequence, committing a model to production when complete, and working on the highest-priority views first.

But overall I think this is a great proposal!

**Comment by dcwalk** [comment]
Thank you for submitting a proposal for Google Summer of Code! 🎉 
I am merging this in as we move into the review phase. Based on [Google's timeline](https://developers.google.com/open-source/gsoc/timeline), on May 4 accepted proposals will be announced

**Pull request merged**